### ISSUE TITLE
Support Synapse Serverless OPENROWSET syntax

### ISSUE THREAD

**Issue by daviewales** [sqlfluff/sqlfluff#5114]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have some code similar to this example code:

``` sql
SELECT *
FROM OPENROWSET(BULK 'https://<storage account>.dfs.core.windows.net/container/folder/*.parquet',
                FORMAT = 'PARQUET') AS [file]
```

When I run `sqlfluff lint`, I get:

```
Found unparsable section: "from OPENROWSET(\n        BULK 'https..."
```

### Expected Behaviour

I expect this syntax to be parsed.

### Observed Behaviour

```
❯ sqlfluff lint --dialect=tsql test.sql
== [test.sql] FAIL
L:   1 | P:   1 | AM04 | Query produces an unknown number of result columns.
                       | [ambiguous.column_count]
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: "FROM
                       | OPENROWSET(\n    BULK 'https://examp..."
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```
sqlfluff lint --dialect=tsql test.sql
```
[test.zip](https://github.com/sqlfluff/sqlfluff/files/12403545/test.zip)

### Dialect

tsql

See [OPENROWSET docs](https://learn.microsoft.com/en-us/azure/synapse-analytics/sql/develop-openrowset#syntax).

### Version

sqlfluff, version 2.3.0
Python 3.10.10

### Configuration

No configuration required to reproduce.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
